Source/WebCore/html/HTMLTextFormControlElement.cpp

```diff
@@ -329,19 +331,20 @@ bool HTMLTextFormControlElement::setSelectionRange(unsigned start, unsigned end,
         // FIXME: Removing this synchronous layout requires fixing setSelectionWithoutUpdatingAppearance not needing up-to-date style.
         protectedDocument()->updateLayoutIgnorePendingStylesheets();
```

Comment:
It seems that this update layout is not necessary when `cacheSelectionIfNotFocusedOrSelected` is true?